server/parsec/components/memory/account.py

```diff
@@ -178,10 +178,8 @@ async def vault_key_rotation(
             case None:
                 return AccountVaultKeyRotation.ACCOUNT_NOT_FOUND
 
-        for account in self._data.accounts.values():
-            for vault in (account.current_vault, *account.previous_vaults):
-                if new_auth_method_id in vault.authentication_methods:
-                    return AccountVaultKeyRotation.NEW_AUTH_METHOD_ID_ALREADY_EXISTS
+        if self.auth_method_id_already_exists(new_auth_method_id):
```

Comment:
BTW can you change `auth_method_id_already_exists` implementation for something similar to the old code here (since the current implementation relies on iterating over each element instead of taking advantage of the fact those elements are keys in a dict)